### DISCUSSION TITLE
Is the `SyncReturn<T>` function executed on the main thread?

### DISCUSSION THREAD

**Discussion by temeddix** [fzyzcjy/flutter_rust_bridge#1028] [Q&A] [answered]
Is the `SyncReturn<T>` function executed on the main Dart thread? Since other Rust functions will be running in the default threadpool of FRB, I was confused about the behavior of this sync function.

There would be two possible scenarios:
- `SyncReturn<T>` function is executed on the main Dart thread
- `SyncReturn<T>` function is executed in the threadpool, blocking the main Dart thread, and when it's finished it returns the calculated value and unblocks the main Dart thread.

Which one is correct? If I get to understand I will contribute to the docs right away.

**Comment by fzyzcjy** [comment]
I have forgotten the details (written more than a year ago and later may be refactored). So there is a simple way to find out: Just print the thread id in wherever you get confused, then we can know whether it is the same thread!

**Comment by temeddix** [comment, marked as answer]
I did an experiment:
<img width="234" alt="image" src="https://user-images.githubusercontent.com/66480156/218104273-d39b6918-243e-4946-b72d-ac46ed5a9d92.png">

Odd lines are `i32` functions. Even lines are `SyncReturn<i32>` functions. Since there are 4 threads in the flutter_rust_bridge pool, obviously `ThreadId(5)` is the main Dart thread. So `SyncReturn<T>` functions ARE using the main Dart thread after all :)
